File: 0120-triangle/0120-triangle.py
class Solution:
  def minimumTotal(self, triangle: List[List[int]]) -> int:
    level, left, right = 0, None, None
    while level < len(triangle)-1:
      level += 1
      for i in range(level+1):
        left = float('inf') if i-1 < 0 else triangle[level-1][i-1]
        right = float('inf') if i >= level else triangle[level-1][i]
        triangle[level][i] += min(left, right)
    return min(triangle[level])

File: 0120-triangle/0120-triangle.java
class Solution {
  public int minimumTotal(List<List<Integer>> triangle) {
    int level = 0;
    int left;
    int right;
    int res;
    while (level < triangle.size() - 1) {
      level++;
      for (int i = 0; i <= level; i++) {
        left = i - 1 < 0 ? 10000000 : triangle.get(level - 1).get(i - 1);
        right = i >= level ? 10000000 : triangle.get(level - 1).get(i);
        res = triangle.get(level).get(i) + Math.min(left, right);
        triangle.get(level).set(i, res);
      }
    }
    return Collections.min(triangle.get(level));
  }
}